### PR TITLE
[Transform] SpinQuant fix OOM

### PR DESCRIPTION
**Pull request by zhanglei1172**
SUMMARY:
"When using SpinQuantModifier for some fuse operations, it is necessary to add the torch.no_grad decorator. Otherwise, PyTorch will capture the grad graph by default, leading to a gradual increase in memory usage. I encountered a CUDA OOM issue when rotating the MOE model, and the OOM problem was resolved after fixing it."

TEST PLAN:
"Performed code quality evaluation locally"

### CODE REVIEW

**Review by kylesayrs — APPROVED** [review]
Awesome catch, thanks!

**Comment by kylesayrs** [comment]
@brian-dellabetta I'm ambivalent towards moving this into a higher scope (calibration pipeline). There might be future modifier techniques which might require gradients (cayley optimization) with which a higher scope would conflict, but those don't exist atm.

**Comment by kylesayrs** [comment]
Either scope would work. Even a lower scope (fuse norms and center embeddings) would work.

**Comment by zhanglei1172** [comment]
@HDCharles I accidentally chose the rebase main; it may need to be approved again.

**Comment by kylesayrs** [comment]
@zhanglei1172 The approvals persist through rebase if there's no conflicts
